client/src/components/Register.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react';
+import { Input } from '@nextui-org/react';
+import { useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import backgroundImage from '../img/register-splash.jpg';
+import { setUser } from '../redux/user/userSlice';
+
+export const Register = () => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const register = async (e) => {
+    e.preventDefault();
+    if (username.length >= 8 && password.length >= 8) {
+      dispatch(setUser({ username, password }));
+      localStorage.setItem('user', JSON.stringify({ username, password }));
+      navigate('/');
+    } else {
+      <p>Username and Password must be at least 8 characters long</p>;
+    }
+  };
```

Comment:
- [ ] OPTIONAL: Your register/login strategy consists of saving a user on localStorage.  This is not the best way to persist users data, because if localStorage is deleted, that user is lost forever and it's impossible to login unless you register another user.  The incognito/private browser tabs won't work either, because their localStorage is deleted on closing.

Also, if I create a user on Firefox, I cannot open that user on another browser like chrome.

Deeming this as optional as proper auth was not specifically requested on the requirements for 4 students.

Reply:
Okay thanks.

Reply:
Skipping this optional requirement due to time constraints. Would work on it before the final presentation. ❌ ❌